lib/src/search/data/repository/locations_repository.dart

```diff
@@ -15,9 +17,11 @@ class LocationsRepository {
   const LocationsRepository(this.api);
   final LocationsApi api;
 
-  Future<Iterable<SearchLocationDto>> getLocations(String search) async {
+  Future<Locations> getLocations(String search) async {
     final result = await api.locations(q: search);
-    final locations = result.map(SearchLocationDto.fromJson);
+    final dto = result.map(SearchLocationDto.fromJson);
+    final places = dto.map((e) => e.toDomain());
+    final locations = Locations(places: places.toList());
```

Comment:
I'd prefer the following syntax, even though it's identical (aka this is just a style issue):

```dart
final locations = Locations(places: [...places]);
```

Reply:
Nevermind, it's not that big of a deal